src/lib/Conversations.ts

```diff
@@ -146,7 +146,7 @@ export default class Conversations<
     const groupsSubscription = XMTPModule.emitter.addListener(
       EventTypes.Group,
       async ({ inboxId, group }: { inboxId: string; group: GroupParams }) => {
-        if (this.known[group.id]) {
+        if (this.known[group.id] || this.client.inboxId !== inboxId) {
```

Comment:
I think this is a good fix to merge now. 

long term, I wonder if it would be a worthwhile performance gain to programmatically generate these Expo Module events to include the inbox-id in their name, so we dont send copies of all clients' events to each client event listener and then need to filter them out. 
https://github.com/xmtp/xmtp-react-native/blob/08341824ca289797405d8494eb553b74113f1b2c/android/src/main/java/expo/modules/xmtpreactnativesdk/XMTPModule.kt#L214-L231

Reply:
I looked into that and it didn't seem straight forward. But something we should consider looking into again for sure.